rotkehlchen/api/v1/resources.py

```diff
@@ -2162,11 +2160,17 @@ def get(self, asset: Asset) -> Response:
 
         return self.rest_api.get_asset_icon(asset, match_header)
 
-    @use_kwargs(upload_schema, location='json_and_view_args')
+
+class AssetIconsResource(BaseMethodView):
+
+    patch_schema = SingleAssetIdentifierSchema()
+    upload_schema = AssetIconUploadSchema()
+
+    @use_kwargs(upload_schema, location='json_and_query')
     def put(self, asset: Asset, file: Path) -> Response:
         return self.rest_api.upload_asset_icon(asset=asset, filepath=file)
 
-    @use_kwargs(upload_schema, location='view_args_and_file')
+    @use_kwargs(upload_schema, location='form_and_file')
```

Comment:
we didn't have json and file and alexey mentioned to reuse this one instead of creating the new one json and file. Also for how files are sent usually you can just add it to the form along the file itself